docusaurus/docs/ui-extensions/register-an-extension.md

```diff
@@ -0,0 +1,128 @@
+- It must be installed and **enabled**.
+
+## How to add an extension to Grafana core/plugins from your app plugin
+
+1. Open the `module.ts(x)` file in your plugin where your `AppPlugin` should be exposed.
+
+1. Use the `configureExtensionLink` on your `AppPlugin` instance to register extensions. This method requires an object containing the following properties:
+
+   - `extensionPointId` *required* - the unique identifier of the extension point you would like to extend.
+   - `title` *required* - used to display your extension at the extension point.
+   - `description` *required* - short description of what your extension does.
+   - `path` - a path within your app plugin where you would like to navigate the user when he/she click the extension.
+   - `onClick` - a callback that should be triggered when the user click the extension.
+   - `category` - a category that we should use to group your extension with other extensions.
+   - `icon` - an icon that should be used while displaying your extension.
+   - `configure` - a function that will be called prior to displaying the extension which enables you to dynamically change/hide your extension depending on the context.
+
+::: note
+
+One of `path` or `onClick` is required. Otherwise the extension will be hidden due to incorrect configuration.
+
+:::
+
+### Example: Adding a regular link
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. When the user clicks the menu item "Go to basic app" he/she will be navigated to `/a/myorg-basic-app/one`.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  extensionPointId: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+});
+```
+
+### Example: Adding a regular link with query string parameters from the context
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. When the user click the menu item "Go to basic app" he/she will be navigated to `/a/myorg-basic-app/one?panelId=12345&timeZone=utc`.
+
+```typescript
+new AppPlugin().configureExtensionLink({
+  title: 'Go to basic app',
+  description: 'Will send the user to the basic app',
+  extensionPointId: 'grafana/dashboard/panel/menu',
+  path: '/a/myorg-basic-app/one', // Must start with "/a/<PLUGIN_ID>/"
+  configure: (context: PanelContext) => {
+    const {timeZone, panelId } = context;
+
+    // You only need to return the properties that you would like to override.
+    return {
+      path: `/a/myorg-basic-app/one?panelId=${panelId}&timeZone=${timeZone}`
+    };
+  }
+});
+```
+
+### Example: Adding a regular link that should be visible depending on the context
+
+In the example below we are adding an extension link to the Grafana dashboard panel menu. It will only be visible for panels with time zone set to UTC.
```

Comment:
```suggestion
In the following example, we add an extension link to the Grafana dashboard panel menu. It will only be visible for panels with the time zone set to UTC.
```